flink-state-backends/flink-statebackend-rocksdb/src/main/java/org/apache/flink/contrib/streaming/state/restore/RocksDBHandle.java

```diff
@@ -94,53 +81,44 @@
     //   - data ingestion after db open: #getOrRegisterStateColumnFamilyHandle before creating
     // column family
-    protected final RocksDbTtlCompactFiltersManager ttlCompactFiltersManager;
-
-    protected RocksDB db;
-    protected ColumnFamilyHandle defaultColumnFamilyHandle;
-    protected RocksDBNativeMetricMonitor nativeMetricMonitor;
-    protected boolean isKeySerializerCompatibilityChecked;
-    protected final Long writeBufferManagerCapacity;
-
-    protected AbstractRocksDBRestoreOperation(
-            KeyGroupRange keyGroupRange,
-            int keyGroupPrefixBytes,
-            int numberOfTransferringThreads,
-            CloseableRegistry cancelStreamRegistry,
-            ClassLoader userCodeClassLoader,
-            Map<String, RocksDbKvStateInfo> kvStateInformation,
-            StateSerializerProvider<K> keySerializerProvider,
-            File instanceBasePath,
+    private final RocksDbTtlCompactFiltersManager ttlCompactFiltersManager;
+
+    private RocksDB db;
+    private ColumnFamilyHandle defaultColumnFamilyHandle;
+    private RocksDBNativeMetricMonitor nativeMetricMonitor;
+    private final Long writeBufferManagerCapacity;
+
+    protected RocksDBHandle(
+            Map<String, RocksDBKeyedStateBackend.RocksDbKvStateInfo> kvStateInformation,
             File instanceRocksDBPath,
             DBOptions dbOptions,
             Function<String, ColumnFamilyOptions> columnFamilyOptionsFactory,
             RocksDBNativeMetricOptions nativeMetricOptions,
             MetricGroup metricGroup,
-            @Nonnull Collection<KeyedStateHandle> stateHandles,
             @Nonnull RocksDbTtlCompactFiltersManager ttlCompactFiltersManager,
             Long writeBufferManagerCapacity) {
-        this.keyGroupRange = keyGroupRange;
-        this.keyGroupPrefixBytes = keyGroupPrefixBytes;
-        this.numberOfTransferringThreads = numberOfTransferringThreads;
-        this.cancelStreamRegistry = cancelStreamRegistry;
-        this.userCodeClassLoader = userCodeClassLoader;
         this.kvStateInformation = kvStateInformation;
-        this.keySerializerProvider = keySerializerProvider;
-        this.instanceBasePath = instanceBasePath;
-        this.instanceRocksDBPath = instanceRocksDBPath;
         this.dbPath = instanceRocksDBPath.getAbsolutePath();
         this.dbOptions = dbOptions;
         this.columnFamilyOptionsFactory = columnFamilyOptionsFactory;
         this.nativeMetricOptions = nativeMetricOptions;
         this.metricGroup = metricGroup;
-        this.restoreStateHandles = stateHandles;
         this.ttlCompactFiltersManager = ttlCompactFiltersManager;
         this.columnFamilyHandles = new ArrayList<>(1);
         this.columnFamilyDescriptors = Collections.emptyList();
         this.writeBufferManagerCapacity = writeBufferManagerCapacity;
     }
 
     void openDB() throws IOException {
+        openDB(Collections.emptyList(), Collections.emptyList());
+    }
+
+    void openDB(
```

Comment:
This is only ever called together with `restoreInstanceDirectoryFromPath()`. Should one of them become an internal method?